test/TestServerProjectsLowLevel/dpg_update1/Custom/ParamsClient.cs

```diff
@@ -0,0 +1,320 @@
+        ///  Overload method to be compatible with dpg_initial where optional parameters become required.
+        /// </summary>
+        /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
+        public virtual async Task<Response> HeadNoParamsAsync(RequestContext context)
+        {
+            using var scope = ClientDiagnostics.CreateScope("ParamsClient.HeadNoParams");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateHeadNoParamsRequest(null, context);
+                return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Head request, no params.
+        ///  Initially has no query parameters. After evolution, a new optional query parameter is added
+        ///  Overload method to be compatible with dpg_initial where optional parameters become required.
+        /// </summary>
+        /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
+        public virtual Response HeadNoParams(RequestContext context)
+        {
+            using var scope = ClientDiagnostics.CreateScope("ParamsClient.HeadNoParams");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateHeadNoParamsRequest(null, context);
+                return _pipeline.ProcessMessage(message, context);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> POST a JSON. Overload method to be compatible with dpg_initial where optional parameters become required. </summary>
+        /// <param name="content"> The content to send as the body of the request. </param>
+        /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
+        /// <remarks>
+        /// Schema for <c>Request Body</c>:
+        /// <code>{
+        ///   url: string (required)
+        /// }
+        /// </code>
+        ///
+        /// </remarks>
+        public virtual async Task<Response> PostParametersAsync(RequestContent content, RequestContext context = null)
+        {
+            Argument.AssertNotNull(content, nameof(content));
+
+            using var scope = ClientDiagnostics.CreateScope("ParamsClient.PostParameters");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreatePostParametersRequest(content, ContentType.ApplicationJson, context);
```

Comment:
Did we decide that our convention is that if we don't take a ContentType, we default ContentType to ApplicationJson?  Why is this better than always taking a ContentType and setting the default value of ContentType to ApplicationJson?

Reply:
This is based on current generated code and I agree that the backward compatible PostParametersAsync method here could cause confusion if a customer never used v1 and started with v2. We can generate the contenType parameter with default value if it only has one content type for LLC.

Reply:
Yes, I see, this was my misunderstanding.

The update1 API looks like this:

```csharp
public virtual Response PostParameters(RequestContent content, RequestContext context = null);
public virtual Response PostParameters(RequestContent content, ContentType contentType, RequestContext context = null);
```

The initial method should make RequestContext be required, since only one overload in a method group should have optional parameters.  Let's discuss in our design meeting today whether contentType should take a default value.

Reply:
Per the discussion, I added `[EditorBrowsable(EditorBrowsableState.Never)]` to the overload methods for v1 that does not have contentType parameter.